### ISSUE TITLE
Add tests for basic business logic on server

### ISSUE THREAD

**Issue by benjaminEwhite** [ncopenpass/CampaignFinance#75]
We should probably have some minimal testing on the server side. As app grows and more folks are contributing, risk of regression increases.

**Comment by SchaeStewart** [comment]
Yeah, I've been neglecting to setup any of this. Do you have any preferred frameworks/methodologies?

I've used [Supertest](https://www.npmjs.com/package/supertest) before with mocks and it works pretty well. In general I dislike using mocks though.

I've also used BDD/Cucumber/Gherkin with a Golang/GraphQL project and that works pretty well IMO.

We can also just add unit tests, but dislike writing unit tests that just mock a bunch of data.

Thoughts?

**Comment by benjaminEwhite** [comment]
Yeah i'm with you re: unit tests on mock data. Actually renamed issue because need not be unit tests.

It's been a while since I've written tests for a Node backend and haven't used Supertest, but skimmed docs and it looks good. I've used Cucumber/Gherkin in the past, but found it to be a bit "too much".

In my opinion, the key thing we should test is the schema for each API endpoint, since our client will be depending on consistency there, and it seems like a place where there's some chance of regression. For instance, if someone edited one of the functions in server/lib/search, it might seem innocent enough but could break the client.  Basically, we just test the "happy path" for each endpoint as a way of ensuring that the API keeps up its side of the contract with the outside world. If the shape of the API response data changes, our test suite should catch that.

Aside from that, I'd incrementally add tests for edge cases or bugs as they come up. For instance, when I discovered that last-name only wasn't yielding candidates ("cooper"), if we had a testing framework in place, I would have added a test for that particular case.

How does all that sit with you @SchaeStewart ?

I think we could get away without having to do mocks. Could use a small subset of the real data as test fixtures.
